This patch fences changes that are only compatible with cups >= 2.2 with
preprocesor directives, restoring compatibility with OS X El Capitan 10.11
while sacrificing said improvements.
--- a/modules/printbackends/cups/gtkprintbackendcups.c
+++ b/modules/printbackends/cups/gtkprintbackendcups.c
@@ -296,10 +296,12 @@ gtk_print_backend_cups_new (void)
   return g_object_new (GTK_TYPE_PRINT_BACKEND_CUPS, NULL);
 }
 
+#if ((CUPS_VERSION_MAJOR == 2 && CUPS_VERSION_MINOR >= 2) || CUPS_VERSION_MAJOR > 2)
 static void                 create_temporary_queue                  (GtkPrintBackendCups *backend,
                                                                      const gchar         *printer_name,
                                                                      const gchar         *printer_uri,
                                                                      const gchar         *device_uri);
+#endif
 
 static void
 gtk_print_backend_cups_class_init (GtkPrintBackendCupsClass *class)
@@ -1872,6 +1874,7 @@ static void
 mark_printer_inactive (GtkPrinter      *printer,
                        GtkPrintBackend *backend)
 {
+#if ((CUPS_VERSION_MAJOR == 2 && CUPS_VERSION_MINOR >= 2) || CUPS_VERSION_MAJOR > 2)
   GtkPrinterCups *cups_printer = GTK_PRINTER_CUPS (printer);
   GList          *iter;
 
@@ -1890,6 +1893,7 @@ mark_printer_inactive (GtkPrinter      *printer,
         }
     }
   else
+#endif
     {
       gtk_printer_set_is_active (printer, FALSE);
       g_signal_emit_by_name (backend, "printer-removed", printer);
@@ -2997,6 +3001,7 @@ cups_create_local_printer_cb (GtkPrintBackendCups *print_backend,
   g_free (printer_name);
 }
 
+#if ((CUPS_VERSION_MAJOR == 2 && CUPS_VERSION_MINOR >= 2) || CUPS_VERSION_MAJOR > 2)
 /*
  *  Create CUPS temporary queue.
  */
@@ -3040,6 +3045,7 @@ create_temporary_queue (GtkPrintBackendCups *backend,
                         NULL,
                         NULL);
 }
+#endif
 
 /*
  *  Create new GtkPrinter from information included in TXT records.
@@ -4519,6 +4525,7 @@ cups_printer_request_details (GtkPrinter *printer)
 
   cups_printer = GTK_PRINTER_CUPS (printer);
 
+#if ((CUPS_VERSION_MAJOR == 2 && CUPS_VERSION_MINOR >= 2) || CUPS_VERSION_MAJOR > 2)
   if (cups_printer->avahi_browsed)
     {
       create_temporary_queue (GTK_PRINT_BACKEND_CUPS (gtk_printer_get_backend (printer)),
@@ -4526,7 +4533,9 @@ cups_printer_request_details (GtkPrinter *printer)
                               cups_printer->printer_uri,
                               cups_printer->temporary_queue_device_uri);
     }
-  else if (!cups_printer->reading_ppd &&
+  else
+#endif
+  if (!cups_printer->reading_ppd &&
            gtk_printer_cups_get_ppd (cups_printer) == NULL)
     {
       if (cups_printer->remote && !cups_printer->avahi_browsed)
